pkg/codegen/schema.go

```diff
@@ -104,6 +104,27 @@ func (p Property) GoTypeDef() string {
 	return typeDef
 }
 
+// GoTypeDefWithOptionalType generates type based on Optional type define here
+// https://github.com/oapi-codegen/runtime/pull/26/files/ ToDo: Change the PR to
+// actual file in repo once it is merged
+func (p Property) GoTypeDefWithOptionalType() string {
```

Comment:
this can actually be collapsed into the existing method. We already have [existing code](https://github.com/deepmap/oapi-codegen/blob/839a661de960d4b11725c46c5a300d66642b0bcc/pkg/codegen/schema.go#L99) that uses global state in code generation, so I'm happy us adding another case in there, instead of introducing a new method